openbb_platform/core/openbb_core/app/static/package_builder.py

```diff
@@ -79,6 +87,48 @@ def create_indent(n: int) -> str:
     return TAB * n
 
 
+class FileLock:
+    """Simple cross-platform file lock wrapper used only for this module."""
+
+    def __init__(self, file_obj):
+        """Initialize the file lock."""
+        self._file = file_obj
+
+    def acquire(self, blocking: bool = True) -> None:
+        """Acquire the file lock."""
+        if _HAS_FCNTL:
+            flags = fcntl.LOCK_EX | (0 if blocking else fcntl.LOCK_NB)
+            fcntl.flock(self._file.fileno(), flags)
+        else:  # Windows via msvcrt
+            import msvcrt as _msvcrt  # pylint: disable=import-outside-toplevel
+
+            mode = _msvcrt.LK_LOCK if blocking else _msvcrt.LK_NBLCK  # type: ignore
+            try:
+                # lock 1 byte at file start; file.seek(0) to ensure position
+                self._file.seek(0)
+                _msvcrt.locking(self._file.fileno(), mode, 1)  # type: ignore
+            except OSError as exc:  # pragma: no cover - platform specific
+                # Normalize to BlockingIOError for parity with fcntl non-blocking
+                raise BlockingIOError from exc
+
+    def release(self) -> None:
+        """Release the file lock."""
+        try:
+            if _HAS_FCNTL:
+                fcntl.flock(self._file.fileno(), fcntl.LOCK_UN)
+            else:
```

Comment:
Same here. Is there a logical branch where `_HAS_FCNTL` is not explicitly set?

Reply:
It's set in a try block, but some of the linters do not recognize that pattern and see it as "using variable before assignment".